src/VASSAL/counters/MovementMarkable.java

```diff
@@ -115,7 +117,15 @@ public String myGetType() {
 
   @Override
   protected KeyCommand[] myGetKeyCommands() {
-    return new KeyCommand[]{new KeyCommand(command, key, Decorator.getOutermost(this), this)};
+    if (kc == null) {  //BR// Bug 13123 - perform proper operation to make sure blank line isn't added to popup menu if command line string is blank
+      kc = new KeyCommand(command, key, Decorator.getOutermost(this), this);
+      if (command.length() > 0 && key != null && ! key.isNull()) {
+        kclist = new KeyCommand[]{kc};
+      } else {
+        kclist = new KeyCommand[0];
+      }
+    }
+    return kclist;
```

Comment:
It's potentially problematic to store these, as `KeyCommand` isn't immutable, so whatever you hand off the key command array to could modify the `KeyCommand` in it, thus modifying the very `KeyCommand` you've kept here to hand out next time.

This way avoids that:
```
return command.isEmpty() || key == null || key.isNull() ?
  new KeyCommand[0] :
  new KeyCommand[]{
    new KeyCommand(command, key, Decorator.getOutermost(this), this)
};
```

Reply:
Sounds good. I was copying the "form" I found in Clone.java but that *totally* doesn't mean it was a good idea :-)  This is clearly better - change added to PR.